enterprise/backend/src/metabase_enterprise/advanced_permissions/models/permissions/block_permissions.clj

```diff
@@ -12,10 +13,15 @@
   current User has unrestricted data permissions from another Group. See the namespace documentation for
   [[metabase.models.collection]] for more details."
   :feature :advanced-permissions
-  [{database-id :database}]
-  (or
-   (not= :blocked (data-perms/full-db-permission-for-user api/*current-user-id* :perms/view-data database-id))
-   (throw (ex-info (tru "Blocked: you are not allowed to run queries against Database {0}." database-id)
-                   {:type               qp.error-type/missing-required-permissions
-                    :actual-permissions @api/*current-user-permissions-set*
-                    :permissions-error? true}))))
+  [{database-id :database :as query}]
+  (let [table-ids (query-perms/query->source-table-ids query)]
```

Comment:
nit: you might not use the `table-ids` value, might be better to move the let around the `(= #{:unrestricted} ...` bit.

Reply:
good call